File: app/src/main/java/ninja/cricks/PhonePeWebViewActivity.java
package ninja.cricks;

import android.os.Bundle;
import android.util.Log;
import android.webkit.WebResourceError;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import ninja.cricks.utils.BindingUtils;

public class PhonePeWebViewActivity extends AppCompatActivity {

    public static String TAG = PhonePeWebViewActivity.class.getSimpleName();

    private WebView webView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        webView = new WebView(this);
        setContentView(webView);

        // Replace "your_phone_pe_url" with the actual URL
        String phonePeUrl = getIntent().getStringExtra(BindingUtils.PHONE_PE_URL);

        webView.getSettings().setJavaScriptEnabled(true);
        webView.setBackgroundColor(0x00000000);

        webView.setWebViewClient(new WebViewClient() {
            @Override
            public void onPageStarted(WebView view, String url, android.graphics.Bitmap favicon) {
                super.onPageStarted(view, url, favicon);
                // Handle page started loading
                Log.e(TAG, "Page started loading: " + url);
            }

            @Override
            public void onPageFinished(WebView view, String url) {
                super.onPageFinished(view, url);
                // Handle page finished loading
                // https://rest.krsdata.net/api/v2/redirectURLPhonePe
                if (url.startsWith("https://rest.krsdata.net/api/v2/redirectURLPhonePe")) {
                    Log.e(TAG, "Response from phone pay: " + url);
                    finish(); // Close the activity when the desired URL is reached
                }
            }

            @Override
            public void onReceivedError(WebView view, WebResourceRequest request, WebResourceError error) {
                super.onReceivedError(view, request, error);
                // Handle page resource error
                Log.e(TAG, "Page resource error code: " + error.getErrorCode() +
                        " description: " + error.getDescription() +
                        " errorCode: " + error.getErrorCode());
            }

            @Override
            public void doUpdateVisitedHistory(WebView view, String url, boolean isReload) {
                super.doUpdateVisitedHistory(view, url, isReload);
                // Handle URL change
                Log.e(TAG, "URL change to " + url);
            }
        });

        webView.loadUrl(phonePeUrl);
    }
}



/*{
  "status": 200,
  "message": "success",
  "response": {
    "amount": 100,
    "user_id": "KUNA2020",
    "redirect_url": "https://rest.krsdata.net/api/v2/redirectURLPhonePe",
    "callback_url": "https://rest.krsdata.net/api/v2/callbackURLPhonePe",
    "order_id": "0202ANUKN1703848013",
    "merchant_id": "NINJA11ONLINE",
    "phonePeKey": "c246fadd-6523-4def-be15-685fc96aa160"
  }
}*/

File: app/src/main/java/ninja/cricks/PhonePeWebViewActivity.kt
package ninja.cricks

import android.content.Context
import android.graphics.Bitmap
import android.os.Bundle
import android.util.Log
import android.view.View
import android.webkit.WebResourceError
import android.webkit.WebResourceRequest
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.isVisible
import androidx.databinding.DataBindingUtil
import ninja.cricks.databinding.ActivityHaodaPaymentBinding
import ninja.cricks.utils.BindingUtils

class PhonePeWebViewActivity : AppCompatActivity() {

    var mBinding: ActivityHaodaPaymentBinding? = null
    var mContext: Context? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_haoda_payment)
        mContext = this
        mBinding!!.apply {
            toolbar.title = intent.getStringExtra(WebActivity.KEY_TITLE)
            toolbar.setTitleTextColor(resources.getColor(R.color.white))
            toolbar.setNavigationIcon(R.drawable.ic_arrow_back_black_24dp)
            setSupportActionBar(toolbar)
            toolbar.setNavigationOnClickListener { v: View? ->
                finish()
            }
            progressBar.visibility = View.VISIBLE

            webBody.settings.javaScriptEnabled = true
            webBody.setBackgroundColor(0x00ffffff)
            webBody.settings.cacheMode = WebSettings.LOAD_NO_CACHE
            webBody.clearCache(true)
        }
        // Replace "your_phone_pe_url" with the actual URL
        val phonePeUrl = intent.getStringExtra(BindingUtils.PHONE_PE_URL)


        mBinding!!.webBody.webViewClient = object : WebViewClient() {

            override fun onPageFinished(view: WebView, url: String) {
                super.onPageFinished(view, url)
                if (mBinding!!.progressBar.isVisible) {
                    mBinding!!.progressBar.visibility = View.GONE
                }
                // Handle page finished loading
                if (url.startsWith("https://rest.ninja11.in/api/v3/paytmCallback")) {
                    Log.e(TAG, "Response from phone pay: $url")
                    finish() // Close the activity when the desired URL is reached
                }
            }

            override fun doUpdateVisitedHistory(view: WebView, url: String, isReload: Boolean) {
                super.doUpdateVisitedHistory(view, url, isReload)
                // Handle URL change
                Log.e(TAG, "URL change to $url")
            }
        }
        mBinding!!.webBody.loadUrl(phonePeUrl!!)
    }

    companion object {
        var TAG = PhonePeWebViewActivity::class.java.simpleName
    }
}